projects/Mallard/src/download-edition/download-and-unzip.ts

```diff
@@ -74,76 +77,81 @@ const runDownload = async (issue: IssueSummary, imageSize: ImageSize) => {
             Feature.DOWNLOAD,
         )
 
-        const issueDataDownload = await downloadNamedIssueArchive({
-            localIssueId: localId,
-            assetPath: assets.data,
-            filename: 'data.zip',
-            withProgress: false,
-        }) // just the issue json
-
-        await issueDataDownload.promise
-
-        await pushTracking('attemptDataDownload', 'completed', Feature.DOWNLOAD)
-
-        await pushTracking(
-            'attemptMediaDownload',
-            JSON.stringify({ localId, assets: assets[imageSize] }),
-            Feature.DOWNLOAD,
-        )
-
-        const imgDL = await downloadNamedIssueArchive({
-            localIssueId: localId,
-            assetPath: assets[imageSize] as string,
-            filename: 'media.zip',
-            withProgress: true,
-        }) // just the images
-
-        await imgDL.promise
-
-        await pushTracking(
-            'attemptMediaDownload',
-            'completed',
-            Feature.DOWNLOAD,
+        await retry(
+            async () => {
+                // We are not asking for progress update during Data bundle download (to avoid false visual completion)
+                // So, instead we are artificially triggering a small progress update to render some visual change
+                updateListeners(localId, {
+                    type: 'download',
+                    data: 0.5,
+                })
+                const dataDownloadResult = await downloadNamedIssueArchive({
+                    localIssueId: localId,
+                    assetPath: assets.data,
+                    filename: 'data.zip',
+                    withProgress: false,
+                })
+                console.log(
+                    'Data download completed with status: ' +
+                        dataDownloadResult.statusCode,
+                )
+
+                await pushTracking(
+                    'attemptDataDownload',
+                    'completed',
+                    Feature.DOWNLOAD,
+                )
+
+                await pushTracking(
+                    'attemptMediaDownload',
+                    JSON.stringify({ localId, assets: assets[imageSize] }),
+                    Feature.DOWNLOAD,
+                )
+
+                const dlImg = await downloadNamedIssueArchive({
+                    localIssueId: localId,
+                    assetPath: assets[imageSize] as string,
+                    filename: 'media.zip',
+                    withProgress: true,
+                })
+                console.log(
```

Comment:
Remove?

Reply:
@jimhunty This seems to be an useful info to know when bundle download fails. Is this going to badly affect sentry quota?

Reply:
No just seemed unnecessary to me. If you see value then keep it.